packages/dht/src/connection/ConnectivityChecker.ts

```diff
@@ -145,20 +157,11 @@ export class ConnectivityChecker {
     }
 
     // eslint-disable-next-line class-methods-use-this
-    private async connectAsync({ host, port, url, timeoutMs, mode }:
-        { host?: string, port?: number, url?: string, timeoutMs: number, mode: ConnectionMode } =
-    { timeoutMs: 1000, mode: ConnectionMode.REQUEST }
+    private async connectAsync({ wsServerInfo, mode, timeoutMs = 1000, }:
+        { wsServerInfo: ConnectivityMethod, mode: ConnectionMode, timeoutMs?: number }
     ): Promise<IConnection> {
         const socket = new ClientWebSocket()
-        let address = ''
-        if (url) {
-            address = url
-        } else if (host && port) {
-            address = 'ws://' + host + ':' + port
-        }
-
-        address += '?' + mode + '=true'
-
+        const address = `${connectivityMethodToWebSocketUrl(wsServerInfo)}?${mode}=true`
```

Comment:
```suggestion
        const url = `${connectivityMethodToWebSocketUrl(wsServerInfo)}?${mode}=true`
```

(also in `ClientWebSocket#connect`)